GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -435,6 +435,38 @@ protected override void OnHandleCreated(EventArgs e)
             base.OnHandleCreated(e);
         }
 
+        public void FixForOwnerScreenLocation(IWin32Window? owner)
+        {
+            if (owner != null)
+            {
+                Screen screenOwner = Screen.FromHandle(owner.Handle);
+                if (!screenOwner.Bounds.IntersectsWith(Bounds))
+                {
+                	// Fix #10121 - Commit form appear out of screen in dual monitor setup
+                	//
+                    // This form appear to be not in the same monitor as owner window
+                    // thus may become invisible when owner is in secondary screen
+                    // with negative left coordinates and last saved position of the
+                    // form was with positive coordinates in case dpi are different.
+                    // Attempting to correct the saved position may be unsuccessful
+                    // because it depends on the screen layout of extended desktop.
+                    if (screenOwner.Bounds.Width - screenOwner.Bounds.X < 0 && Left > 0)
```

Comment:
(actual change seem reasonable, have not tried to get all details)

Reply:
Could you explain why a handling is required only in this case, please? I think it should be `<=` and `>=` at least - perhaps even `if (screenOwner.Bounds.X < 0 && Left >= 0)`.
What if `screenOwner.Bounds.X + screenOwner.Bounds.Width` is exceeded?